.github/workflows/python_bindings.yml

```diff
@@ -102,6 +102,13 @@ jobs:
           echo "$MSVC_DIR\VC\Auxiliary\Build" >> $GITHUB_PATH
         fi;
 
+        if [[ "$GITHUB_REF" == *"refs/tags"* ]]; then
+          PYTHON_PIP_REPOSITORY=pypi
+        else
+          PYTHON_PIP_REPOSITORY=testpypi
+        fi;
+        echo "PYTHON_PIP_REPOSITORY=$PYTHON_PIP_REPOSITORY" >> $GITHUB_ENV
```

Comment:
So I'm not 100% sure this is going to work in all cases... There is one path where it would fail:
* Our project is at 3.3.0.
* We compute the next available version by looking at pypi, where it doesn't exist yet. so 3.3.0 is selected.
* We upload to testpyi with this version, and it works

* In the event the push to testpypi works BUT the bindings don't during testing, it is NOT uplaod to pypi
* We rerun the workflow: we still compute 3.3.0 because we looked at pypi which still doesn't have it
* Now the push to testpypi won't work since it already exists

Reply:
Perhaps the easiest way would be to build the python package twice with the testpypi calculated version and the pypi calculated one...

Reply:
or, in the event we do a release, don't even bother trying to upload to testpypi, just upload as an artifact and download that directly in the testing stage

Reply:
Hmm, good point. What about upload the bindings to testpypi using the normal sequence by appending the test version(e.g. 3.2.2a4, 3.2.2a5), test those packages, and then add another step to change the name of the bindings version to the official release  (3.2.2) and upload those?  Then upload those to pypi. 

Another option would be if the testpypi version doesn't pass the tests we yank the version?  I don't love that idea but it's also test server so maybe this okay to do this. Not sure if we can do this with cli calls.